source/MRMesh/MRObjectMeshHolder.h

```diff
@@ -130,7 +130,7 @@ class MRMESH_CLASS ObjectMeshHolder : public VisualObject
 
     /// the texture ids for the faces if more than one texture is used to texture the object
     /// texture coordinates (uvCoordinates_) at a point can belong to different textures, depending on which face the point belongs to
-    virtual void setTexturePerFace( Vector<TextureId, FaceId> texturePerFace ) { texturePerFace_ = std::move( texturePerFace ); dirty_ |= DIRTY_TEXTURE_PER_FACE; }
+    virtual void setTexturePerFace( Vector<TextureId, FaceId>& texturePerFace ) { std::swap( texturePerFace_, texturePerFace ); dirty_ |= DIRTY_TEXTURE_PER_FACE; }
```

Comment:
better add function `updateTexturePerFace`, it is unexpected that set function performs swap